### PR TITLE
Unify counting of image layers in our helper and registry

### PR DESCRIPTION
**Pull request by legionus**
Unify counting of image layers in our helper and registry. The registry counts size of each unique layer just once.

@miminar @soltysh @mfojtik

### CODE REVIEW

**Comment by legionus** [comment]
[test]

**Comment by miminar** [comment]
Awesome! A unit test would make this perfect. But I'm fine with merging this as is.

**Comment by soltysh** [comment]
I'd say add a test case and I'll merge it.

**Comment by legionus** [comment]
@miminar @soltysh I found another serious bug: `DockerImageConfig` not parsed in the right way. Fixed. Added test checks both.

**Comment by legionus** [comment]
[test]

**Comment by soltysh** [comment]
LGTM, will merge once tests are green.

**Comment by legionus** [comment]
re[test]

**Comment by soltysh** [comment]
This is failing on:
```
FAILURE after 0.916s: test/cmd/admin.sh:309: executing 'oadm registry -o yaml --credentials=/tmp/openshift/test-cmd//openshift.local.home/.kube/non-default-config' expecting success and text 'image:.*\-docker\-registry': the output content test failed
```
which is weird, lemme re-run tests.

re-[test]

**Comment by legionus** [comment]
[test]

**Comment by soltysh** [comment]
[merge]

File: pkg/image/api/helper_test.go [hunk 1]
```diff
@@ -741,6 +879,85 @@ func TestImageWithMetadata(t *testing.T) {
...
+				DockerImageMetadata: DockerImage{
+					ID:            "sha256:815d06b56f4138afacd0009b8e3799fcdce79f0507bf8d0588e219b93ab6fd4d",
+					Parent:        "",
+					Comment:       "",
+					Created:       unversioned.Date(2015, 2, 21, 2, 11, 6, 735146646, time.UTC),
+					Container:     "e91032eb0403a61bfe085ff5a5a48e3659e5a6deae9f4d678daa2ae399d5a001",
+					DockerVersion: "1.9.0-dev",
+					Author:        "",
+					Architecture:  "amd64",
+					Size:          882848,
```
Comment (miminar) [inline pkg/image/api/helper_test.go]: Have you modified the config in some way? If I count the layer sizes and size of image config as listed in the manifest, I get:
```
639152 + 235231 + 5312 + 7023 = 886718
```

I can get to `882848` if I don't look at image manifest but rather count the size of the config manually.

**Pull request merged**